### ISSUE TITLE
International Keyboard Issues - Wont Type or Copy and Paste International Characters

### ISSUE THREAD

**Issue by dieskim** [koush/vysor.io#664]
**Describe the bug**
International Keyboard Issues - Wont Type or Copy and Paste International Characters

This only happens some of the time on the device. Its even happened when I have 4 devices plugged in and viewing on Vysor, one device wont be able to Type or Copy Paste International Characters while the other devices work fine.

At times a simple "Vysor Reload" fixes this - other times the phone and Mac need to be completely restarted.

**Attach a Vysor Bug Report Log**
Can provide Gist link in Private - contains personal info.
The device having the issue:
id 108521476768516801745

**To Reproduce**
1. Open Vysor Application
2. Enable International Keyboard
3. Click "View" to open device
4. Set Mac OSX to International Keyboard
5. Try to type international characters - FAIL - nothing is sent to device
6. Try to copy and paste international characters - FAIL - nothing is pasted on device
7. Change Mac OSX to English Keyboard
8. Try to type english characters - WORKS
9. Try to copy and paste english characters - WORKS
10. Change Mac OSX back to International Keyboard
11. Try to type international characters - FAIL - nothing is sent to device
12. Try to copy and paste international characters - FAIL - nothing is pasted on device

**Expected behavior**
International Keyboard option should allow typing and copy pasting international characters at all times

**Desktop (please complete the following information):**
 - OS:  Mac OSX 
 - Vysor Version 3.1.4

**Android (please complete the following information):**
 - Device: Samsung Galaxy J4+
 - OS: Android 8.0.0
 - Vysor Bug Report Link: Can provide Gist link in Private - contains personal info.

**Android Logcat**
None

**Additional context**
None

**Comment by dieskim** [comment]
@koush I have not installed any custom ADB versions or anything - what version of ADB for Mac OSX would be best? I am wondering if an ADB restart would help when a Vysor Reload does not? 

Could it also be possibly that there is some type of ADB Queue (not sure what to call it) issue where the queue is full?
This does seem to happen when I copy paste alot of content for a while, the other device language input might even slow down and then speed up again.

**Comment by dieskim** [comment]
@koush - just an update regarding this when it happened again today. 
It seems like there are two separate symptoms, not sure if they are related.

1. At times all input via the "International Keyboard" option will not work - both typing and pasting. This would be for example when setting the Mac OSX keyboard to an International Keyboard and trying to type / paste on the Android Phone. In this instance you can change the Mac OSX keyboard to English and all works fine.
This problem is only solved via a full Vysor and Device restart.

2. At times the input slows down dramatically when using the "International Keyboard" - typing would be extremely delayed - type a few characters and they come up very slow - pasting would also be very slow and most of the time only a few characters would be pasted - not the whole string. 
This problem seems to solve itself if you just leave it a bit - after a few minutes it might go back to normal

**Comment by dieskim** [comment]
@koush - another update on this. It just happened again. This time I was able to learn two more things
1. When the International Keyboard input is not working, latin characters work just fine. When you type things like a space (" "), full stop "." , those all are sent over just fine - just the non latin characters are not sent over.
2. After trying many things to get the device to type international characters again I was finally able to get it working via a "Force Stop" on the App on the phone itself. Once I did a "Force Stop" I then had to waif for it so start up again and then set my keyboard back to the Vysor Keyboard - then it could type just fine.
Things I tried before this was
- close the Vysor View of the Device
- Reload Vysor
- unplug and then reconnect the Device
- change keyboards to default Samsung Keyboard and back to Vysor
all of this did not work then finally did a "Force Stop" of the Vysor app, and that worked.

This was reproducible on other devices as well when they did the same
